scio-examples/src/main/scala/com/spotify/scio/examples/extra/SortMergeBucketExample.scala

```diff
@@ -154,3 +152,47 @@ object SortMergeBucketJoinExample {
     ()
   }
 }
+
+object SortMergeBucketTransformExample {
+  import com.spotify.scio.smb._
+
+  def main(cmdLineArgs: Array[String]): Unit = {
+    val (sc, args) = ContextAndArgs(cmdLineArgs)
+    val inputL = args("inputL")
+    val inputR = args("inputR")
+    val output = args("output")
+
+    implicit val coder: Coder[GenericRecord] =
+      Coder.avroGenericRecordCoder(SortMergeBucketExample.UserDataSchema)
+
+    sc.sortMergeTransform(
```

Comment:
Is there a point to break/curry this into `sc.sortMergeTransform(in1, in2).with(fn).into(out)`, and reuse some bits for 3 & 4 way cases?

Reply:
refactored the api a bit -- can you take a look? I think it's ready for merge